helix-core/src/main/java/org/apache/helix/monitoring/mbeans/dynamicMBeans/DynamicMBeanProvider.java

```diff
@@ -157,6 +157,15 @@ public synchronized void unregister() {
     _objectName = null;
   }
 
+  /**
+   * Check whether the MBean is registered or not.
+   *
+   * @return True, if the MBean is already registered. Otherwise, false.
+   */
+  public boolean isRegister() {
+    return _objectName != null;
```

Comment:
This is not logically correct. But we should always check it through MBeanServer, that is the source of truth.
Adding this check depends on our internal assumption is dangerous.

Reply:
Removed.